lib/internal/inspector_async_hook.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const { createHook } = require('async_hooks');
+const inspector = process.binding('inspector');
+
+if (!inspector || !inspector.asyncTaskScheduled) {
+  exports.setup = function() {};
+  return;
+}
+
+const hook = createHook({
+  init(asyncId, type, triggerId, resource) {
```

Comment:
nit: please use `triggerAsyncId` as the parameter name (instead of `triggerId`). We have been really bad at making the naming consistent and are now trying to correct that. Sorry for the confusion.